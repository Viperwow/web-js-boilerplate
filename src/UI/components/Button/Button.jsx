import {identity as _identity} from 'lodash';
import React from 'react';

const Button = ({
  text,
  t = _identity,
}) => (
  <button type="submit" className="btn">
    {t(text)}
  </button>
);

export default Button;
