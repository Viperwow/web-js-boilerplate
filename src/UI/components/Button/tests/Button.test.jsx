import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Button from '../Button';

jest.unmock('../Button');

describe('Button component should render correctly', () => {
  test('With default props', () => {
    const output = shallow(<Button />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });

  test('With text prop', () => {
    const output = shallow(<Button text="text" />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });
});
