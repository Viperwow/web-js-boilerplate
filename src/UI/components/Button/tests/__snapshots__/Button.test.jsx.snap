// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly With default props 1`] = `
<button
  className="btn"
  type="submit"
/>
`;

exports[`Button component should render correctly With text prop 1`] = `
<button
  className="btn"
  type="submit"
>
  text
</button>
`;
